### PR TITLE
fix(kuma-cp): consumer scoped policies should be applied only on dpps from the same namespace

### PR DESCRIPTION
**Pull request by Automaat**
Without this change, consumer policy is matched to all dpps in zone no matter the namespace, which is wrong as it should be namespace scoped

### Checklist prior to review

- [x] [Link to relevant issue][1] as well as docs and UI issues --
- [x] This will not break child repos: it doesn't hardcode values (.e.g "kumahq" as a image registry) and it will work on Windows, system specific functions like `syscall.Mkfifo` have equivalent implementation on the other OS --
- [x] Tests (Unit test, E2E tests, manual test on universal and k8s) --
  - Don't forget `ci/` labels to run additional/fewer tests
- [x] Do you need to update [`UPGRADE.md`](../blob/master/UPGRADE.md)? --
- [x] Does it need to be backported according to the [backporting policy](../blob/master/CONTRIBUTING.md#backporting)? ([this](https://github.com/kumahq/kuma/actions/workflows/auto-backport.yaml) GH action will add "backport" label based on these [file globs](https://github.com/kumahq/kuma/blob/master/.github/workflows/auto-backport.yaml#L6), if you want to prevent it from adding the "backport" label use [no-backport-autolabel](https://github.com/kumahq/kuma/blob/master/.github/workflows/auto-backport.yaml#L8) label) --

[1]: https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword

### CODE REVIEW

File: pkg/plugins/policies/core/matchers/dataplane.go [hunk 1]
```diff
@@ -171,6 +174,15 @@ func dppSelectedByPolicy(
 	}
 }
 
+func ddpSelectedByNamespace(meta core_model.ResourceMeta, dpp *core_mesh.DataplaneResource) bool {
+	if meta.GetLabels()[mesh_proto.PolicyRoleLabel] == string(mesh_proto.ConsumerPolicyRole) {
```
Comment (lobkovilya) [inline pkg/plugins/policies/core/matchers/dataplane.go:178]: should be both consumer and workload-owner https://github.com/kumahq/kuma/blob/180976f81dcb6a9b86c4c3fa534e11534060171d/docs/madr/decisions/052-policy-on-namespace.md?plain=1#L103-L102

File: pkg/plugins/policies/core/matchers/dataplane.go [hunk 2]
```diff
@@ -129,6 +129,9 @@ func dppSelectedByPolicy(
 ) ([]core_rules.InboundListener, []core_rules.InboundListenerHostname, bool, error) {
 	switch ref.Kind {
 	case common_api.Mesh:
+		if !ddpSelectedByNamespace(meta, dpp) {
```
Comment (lobkovilya) [inline pkg/plugins/policies/core/matchers/dataplane.go:132]: it's not just for kind Mesh, it should be checked for all kinds no?

File: pkg/plugins/policies/core/matchers/dataplane.go (hunk 1 above)
Comment (Automaat) [reply pkg/plugins/policies/core/matchers/dataplane.go]: changed

File: pkg/plugins/policies/core/matchers/dataplane.go (hunk 2 above)
Comment (Automaat) [reply pkg/plugins/policies/core/matchers/dataplane.go]: you are right it applies to all types, changed

**Review by lobkovilya — APPROVED** [review]

**Pull request merged**